### DISCUSSION TITLE
Desmos Graphing Calculator Integration

### DISCUSSION THREAD

**Discussion by RobertClarke64** [outline/outline#5011] [Ideas]
I'd love if Outline had integration with Desmos Graphing calculator. I'd like to plot graphs using maths equations, preferably keeping them interactive in Outline, rather than just screenshots.

Desmos has an [embed API](https://www.desmos.com/api/v1.7/docs/index.html#document-quickstart), which allows you to very quickly and simply add an interactive graph to your web page. This API also allows you to add observers, as well as getting the state of the graph in order to save when edited.

![image](https://user-images.githubusercontent.com/26649481/224092211-c6d33b85-bafe-4520-ae99-79a0c78b0542.png)

I'm not sure how it will work with API keys when in production. An API key must be requested from Desmos to use it. They also allow self-hosting of their API, by giving access to their private GitHub repository. This would obviously mean that anybody self-hosting Outline and wanting to use Desmos would need to either request an API key, or request to self-host the API. A field for the Desmos API key could obviously be added to Outline's preferences though.

I think this would be a brilliant addition to Outline. I'd like to create a pull request myself, but I'm unfamiliar with Outline's codebase, and need to keep up with my university work currently.

**Comment by xhyrom** [comment]
Is there any way that we'll get this? I would really appreciate it :)
